Add IQS046 rule (#2420)
home-assistant/developers.home-assistant: docs/core/integration-quality-scale/rules/iqs046.md

---

> title: "IQS046 - The documentation provides a list of known supported / unsupported devices"

Reviewer: ```suggestion
title: "IQS046 - The integration documents known supported / unsupported devices"
```

---

IQS046 is as follows:

> The documentation provides a list of known supported / unsupported devices

Reviewer: ```suggestion
The integration documents known supported / unsupported devices
```

## Reasoning

A lot of Home Assistant users buy devices based on if Home Assistant supports them.

> To make it easier for users to find out if a device is supported, the documentation should provide a list of known supported or unsupported devices.

Reviewer: ```suggestion
To make it easier for users to find out if a device is supported, the documentation should document the known supported or unsupported devices.
```

> To make it easier for users to find out if a device is supported, the documentation should provide a list of known supported and unsupported devices.

Reviewer: I don't think we need to list both supported and unsupported devices unless the situation requires that. It may be possible to assume that all devices from a manufacturer are supported except the listed unsupported ones and vice versa. Each integration can decide what is needed to describe the situation.

Listing explicitly all devices from a manufacturer if all are known to be supported doesn't seem needed. A sentence saying that all devices are supported would be enough, I think.
```suggestion
To make it easier for users to find out if a device is supported, the documentation should provide a list of known supported or unsupported devices.
```

Author: That is a good point, but I do think that having some sort of consistency would be nice. It would also increase confidence of users if they find that their device is supported, where as of now, it feels more like a gamble. (and I think it would be required to have some sort of disclaimer telling that this can't be 100% accurate)

Reviewer: I don't think we always need to have a header for supported devices and a header for unsupported devices. That would just become boilerplate for many integrations. It's enough that we say that integrations should document this.

> This would decrease the amount of bad experiences where the user finds out their device is not supported when they try to set it up.

Reviewer: ```suggestion
This will decrease the amount of bad experiences where the user finds out their device is not supported when they try to set it up.
```

## Example implementation

```markdown
## Supported devices

The following devices are known to be supported by the integration:
- Device 1
- Device 2
- Every appliance that runs MyOS

## Unsupported devices

The following devices are not supported by the integration:
- Device 3
- Appliances built before 2010
```

## Additional resources

## Exceptions

This rule does not apply to integrations that do not connect to a device or service.

> This rule also does not apply to integrations that don't expose real life physical devices.

Author: So what I try to mean with this exception, is that integrations like `nyt_games` or `plex` or `tautulli` don't expose real life devices to Home Assistant. They just provide services, not physical devices.

But in the case of Withings, where we receive data on user level, not on device level, I do want to apply the rule because withings also produces cameras nowadays for some reason. Like, I do want to give a list of devices to tell the users, all these health devices are supported, and when new ones come out, I want to check if I implemented everything that that device provides so I can assure users that its usable.

Reviewer: ```suggestion
This rule also does not apply to integrations that don't integrate physical devices.
```

Reviewer: ```suggestion
This rule also does not apply to integrations that don't integrate physical devices.
```
